src/test/java/com/example/springbootintro/controller/BookControllerTest.java

```diff
@@ -0,0 +1,218 @@
+    }
+
+    @Test
+    @WithMockUser(username = "admin", authorities = "ADMIN")
+    @Sql(scripts = "classpath:database/book/add-new-book-to-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/book/remove-books-from-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void deleteById_deleteBookById_ok() throws Exception {
+        Long bookId = 1L;
+        mockMvc.perform(delete("/api/books/" + bookId))
+                .andExpect(status().isOk());
+
+        MvcResult result = mockMvc.perform(
+                        get("/api/books/")
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        String contentAsString = result.getResponse().getContentAsString();
+        List<BookResponseDto> actual = List.of(
+                objectMapper.readValue(contentAsString, BookResponseDto[].class));
+
+        Assertions.assertEquals(actual.size(), 0);
+    }
+
+    @Test
+    void deleteById_deleteBookByIdUnauthorizedUser_notOk() throws Exception {
+        Long bookId = 1L;
+        mockMvc.perform(delete("/api/books/" + bookId))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser(username = "admin", authorities = "ADMIN")
+    @Sql(scripts = "classpath:database/book/add-new-book-to-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/book/remove-books-from-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void update_updateBookById_ok() throws Exception {
+        Long bookId = 1L;
+        BookRequestDto testBook = createABook();
+        String jsonObject = objectMapper.writeValueAsString(testBook);
+
+        mockMvc.perform(put("/api/books/" + bookId)
+                        .content(jsonObject)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(1L))
+                .andExpect(jsonPath("$.title").value(testBook.getTitle()))
+                .andExpect(jsonPath("$.author").value(testBook.getAuthor()))
+                .andExpect(jsonPath("$.isbn").value(testBook.getIsbn()))
+                .andExpect(jsonPath("$.description").value(testBook.getDescription()));
+    }
+
+    @Test
+    void update_updateBookUnauthorizedUser_notOk() throws Exception {
+        Long bookId = 1L;
+        BookRequestDto testBook = createABook();
+        String jsonObject = objectMapper.writeValueAsString(testBook);
+        mockMvc.perform(put("/api/books/" + bookId)
+                        .content(jsonObject)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isUnauthorized());
+    }
+
+    private BookRequestDto createABook() {
```

Comment:
articles are usually not written in names 
 ```suggestion
    private BookRequestDto createBook() {
```